Assets/MixedRealityToolkit/_Core/Managers/MixedRealityManager.cs

```diff
@@ -4,6 +4,8 @@
 using Microsoft.MixedReality.Toolkit.Internal.Definitions;
 using Microsoft.MixedReality.Toolkit.Internal.Interfaces;
 using System;
+using System.Linq;
```

Comment:
Tisk tisk. Any way to remove this? Or are we using a Linq specific method?

Reply:
@StephenHodgson , I don't see anything in the coding standards doc regarding not using Linq. Should there be; perhaps with an actual reason  as to why it shouldn't be used?

Reply:
Actually there is, in a few places.  Give me a few min to find the relevant links (get it? 😆 )

Reply:
I opened it and searched for "linq" and came up empty.

Reply:
[Windows Mixed Reality Developer Documentation:](https://docs.microsoft.com/en-us/windows/mixed-reality/performance-recommendations-for-unity#garbage-collections)

>Tips for avoiding allocations. The ideal way to avoid allocations (and thus collections) is to allocate everything you need at the startup of the application and just reuse that data while the app runs. There are plenty of cases where it's not possible to avoid allocations (for example APIs that return allocated objects), so this isn't actually possible in practice, but it's a good goal to shoot for.
>
>There are also some unintuitive reasons why allocations may occur even when you're not directly allocating something, here are some tips to avoid that:
>
> - **Do not use LINQ, as it causes heavy allocations.**
> - Do not use lambdas, as they cause allocations.
> - Beware of boxing! A common case for that is passing structs to a method that takes an interface as a parameter. Instead, make the method take the concrete type (by ref) so that it can be passed without allocation.
> - Prefer structs to classes whenever you can.
> - Default implementations for value equality and GetHashcode uses reflection, which is not only slow but also performs a lot of allocations.
> - Avoid foreach loops on everything except raw arrays and List. Each call potentially allocates an Enumerator. Prefer regular for loops whenever possible. (See https://jacksondunstan.com/articles/3805 for more info)

Reply:
Maybe that should be called out explicitly in the [MRTK coding standards](https://github.com/Microsoft/MixedRealityToolkit-Unity/blob/MRTK-Version-Next/CodingGuidelines.md) document as well. Thoughts? What about the next line regarding lambdas?

Reply:
Yes, we should probably align our coding standards to point and quote the Mixed Reality Documentation.

Reply:
I'll use lambda's but only when necessary and when we start getting into async code that perf stuff won't really be as big of a deal.

Reply:
Any time I see a comment like the one at the top of this thread, I'd like to see a reason behind the suggestion. It will help future contributors understand better the logic and thinking behind the guidelines.

Reply:
Understandable

Reply:
I've removed ALL bar one implementation.  Because there just isn't another clean way to Sort :D

Reply:
That's fine with me. It's only called once on startup anyway.